developing-without-a-build/es-dev-server.md

```diff
@@ -0,0 +1,195 @@
+
+It would be a shame if we have to run a production build every time we want to run our app on one of these browsers. `es-dev-server` supports multiple compatibility modes that help with this.
+
+When compatibility mode is enabled, the server handles the necessary polyfills and code transformations for older browsers. This takes us into build tooling territory a little bit, so we're no longer purely "developing without build tools". I think that's fine, as we're using it specifically for browser compatibility. You have to opt-in to this behavior explicitly.
+
+Let's see how it works in action. Add a dynamic import to `app.js` to lazy load a module when a button is clicked:
+
+```js
+import { html, render } from 'lit-html';
+
+async function lazyLoad() {
+  await import('lit-html/directives/until.js');
+}
+
+const template = html`
+  <button @click=${lazyLoad}>Click me!</button>
+`;
+render(template, document.getElementById('app'));
+```
+
+This is a silly example, as the dynamic import does not really do anything functional. If we run this as-is on Chrome, Safari, and Firefox this works just fine. However, Edge does not yet support dynamic imports so this code would not run there.
+
+We can turn on the lightest compatibility mode, `esm` to handle this case:
+```bash
+npx es-dev-server --node-resolve --compatibility esm --open
+```
+
+With `esm` enabled, `es-dev-server` injects [es-module-shims](https://www.npmjs.com/package/es-module-shims) and adds a loader script to your index. You don't need to change any of your code for this. You can view the injected script in the index file:
+
+![](./images/esm-compat.png)
+
+There is some extra boilerplate, as the other compatibility modes might add more polyfills. The polyfills are hashed so that they can be cached aggressively in the browser.
+
+Besides `esm` there are the `modern` and `all` compatibility modes. These modes inject polyfills for common browser APIs and use `@babel/preset-env` for transforming the latest javascript syntax to a compatible format.
+
+In `modern` compatibility mode you can write modern code and run it on the latest two versions of Chrome, Safari, Firefox, and Edge.
+
+In `all` compatibility mode support is extended to older browsers, all the way to IE11. On browsers which don't support es modules, they are transformed to [systemjs modules](https://www.npmjs.com/package/systemjs).
+
+## Import maps
+In the previous article, we briefly discussed import maps as an upcoming browser feature which handles bare module imports. `es-module-shims` and `systemjs` both support import maps. If compatibility mode is enabled, the dev server takes care of adapting your import maps to work with these libraries, making this a viable workflow during development.
```

Comment:
Would be nice to have an example of how to use import maps as well

Reply:
It feels a bit outside of the scope of this article, do we have some other article to link to?

Reply:
Sure, sounds good